extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/HttpSecurityImpl.java

```diff
@@ -70,6 +74,25 @@ public HttpSecurity mechanism(HttpAuthenticationMechanism mechanism) {
                 throw new IllegalArgumentException("Cannot configure basic authentication programmatically because "
                         + "the authentication realm has already been configured in the 'application.properties' file");
             }
+        } else if (mechanism.getClass() == MtlsAuthenticationMechanism.class) {
+            boolean mTlsEnabled = !ClientAuth.NONE.equals(clientAuth);
```

Comment:
I believe `ClientAuth.NONE` should be rejected when the user calls `mTLS(ClientAuth.NONE)` as it is already conradictory